pkg/service/verifycredential/verifycredential_service.go

```diff
@@ -233,3 +242,40 @@ func (s *Service) ValidateVCStatus(vcStatus *verifiable.TypedID, issuer string)
 
 	return nil
 }
+
+func (s *Service) parseAndVerifyVCSDJWT(combinedFormatForPresentationBytes []byte) (*verifiable.Credential, error) {
+	sdJwtCombinedFormatForPresentation := string(vcsverifiable.UnQuote(combinedFormatForPresentationBytes))
+	disclosedClaims, err := verifier.Parse(sdJwtCombinedFormatForPresentation,
+		verifier.WithSignatureVerifier(
+			jwt.NewVerifier(jwt.KeyResolverFunc(
+				verifiable.NewVDRKeyResolver(s.vdr).PublicKeyFetcher(),
+			))),
+		verifier.WithIssuerSigningAlgorithms(vcsverifiable.SDJWTSignatureTypes))
+	if err != nil {
+		return nil, fmt.Errorf("SD-JWT verifier.Parse error: %w", err)
+	}
+
+	// Get combined format for presentation.
+	cfp := common.ParseCombinedFormatForPresentation(sdJwtCombinedFormatForPresentation)
+
+	// Parse credential from SD-JWT.
+	cred, err := verifiable.ParseCredential([]byte(cfp.SDJWT),
```

Comment:
Why we cannot use disclosedVC (you call it disclosedClaims) to unmarshal VC from? Why do you need to ParseCredential from cfp.SDJWT?